src/app/locator/qgsinbuiltlocatorfilters.cpp

```diff
@@ -308,16 +311,24 @@ QStringList QgsActiveLayerFeaturesLocatorFilter::prepare( const QString &string,
     if ( field.configurationFlags().testFlag( QgsField::ConfigurationFlag::NotSearchable ) )
       continue;
 
-    if ( !_fieldRestriction.isEmpty() && !field.name().startsWith( _fieldRestriction ) )
+    if ( !_fieldRestriction.isNull() && !field.name().startsWith( _fieldRestriction ) )
```

Comment:
why this change? it's a step back, in that the concept of null QStrings is deprecated...

Reply:
I wanted to distinguish when '@' was typed with nothing yet. But I can replace it with a `startsWith('@')`.

Reply:
Why doesn't testing .isEmpty() work for the same result?

Reply:
My first idea was:
search: 'something' => fieldRestriction is null
search: '@field something' => fieldRestriction is field
search: '@' => fieldRestriction is empty string, not null

I added now a pointer to a boolean in the private method, things are a bit more explicit like this.